### PR TITLE
Fix Conditional Typing in useSWRMutation to Allow Optional ExtraArg Without Explicitly Passing Undefined

### PR DESCRIPTION
**Pull request by saengmotmi**
This PR contains the fixes for the issue: https://github.com/vercel/swr/issues/2665.

This is my first contribution to SWR. Thank you in advance :)

### CODE REVIEW

File: mutation/types.ts [hunk 1]
```diff
@@ -56,7 +56,7 @@ interface TriggerWithArgs<
   ExtraArg = never
 > {
   <SWRData = Data>(
-    extraArgument: ExtraArg,
+    extraArgument?: ExtraArg,
```
Comment (promer94) [inline mutation/types.ts:59]: I don't think this is the right fix. And it breaks the type validation for required `args`

What we expect here are 

For required `arg`

```tsx
const { trigger, reset, data, error } = useSWRMutation(
    '/api/user',
    (_, opts: { arg: number }) => String(opts.arg)
)
// should throw error, but your change breaks this.
trigger()
// pass type validation
trigger(1)
```

For optional `arg`

```tsx
const { trigger, reset, data, error } = useSWRMutation(
    '/api/user',
    (_, opts: { arg?: number }) => String(opts.arg)
)
// pass type validation
trigger()
// pass type validation
trigger(1)
```

File: test/type/trigger.ts [hunk 2]
```diff
@@ -94,9 +94,6 @@ export function useTestSWRMutation() {
   )
 
   const test = () => {
-    // @ts-expect-error `arg` should be 'foo'
```
Comment (promer94) [inline test/type/trigger.ts:97]: This part is actually relevant, you shouldn't remove them

**Review by promer94 — CHANGES_REQUESTED** [review]
I think the correct way to fix this problem is to have a new `TriggerWithOptionsArgs` type. 

So you can improve the type inference implementation here.
https://github.com/vercel/swr/blob/7badbb9d428d2ac48873b3a3f11d54e7cec0c7cf/mutation/types.ts#L142-L144

`ExtraArg = never`  ->  `TriggerWithoutArgs`
`ExtraArg = number | undefined` -> `TriggerWithOptionsArgs`
`ExtraArg = number` -> `TriggerWithArgs`


Also. You need to add a new test for optional `arg`

**Comment by saengmotmi** [comment]
@promer94 Oh, I see. I had made incorrect changes. I understand your suggestion. I've made changes according to your suggestion.

I introduced a new `TriggerWithOptionsArgs` type and made the appropriate adjustments for type inference. I have also added a new test for the optional argument.

Please review my latest changes and let me know if there's anything else I need to address.

**Review by promer94 — APPROVED** [review]
Thanks!

**Pull request merged**